### DISCUSSION TITLE
Experimental Features

### DISCUSSION THREAD

**Discussion by kylecorry31** [kylecorry31/Trail-Sense#2099] [Announcements]
New experimental features will be posted here. You can enable the experimental features under Settings > Experimental.

Note that experimental features are not ready for everyday use, and may contain bugs. You can report bugs, suggest features, or ask for help about experimental features here.

**Comment by kylecorry31** [comment]
# Augmented Reality
The Augmented Reality tool allows you to use your phone's sensors and camera to see beacons or the path of the sun / moon overlayed onto the real world. 

## Availability
Available on all devices that have a compass sensor. If your device does not have a gyroscope, the experience won't be great, but is still possible.

## Planned Features and Known Bugs
[Search issues with the label of 'augmented-reality'](https://github.com/kylecorry31/Trail-Sense/issues?q=is%3Aopen+is%3Aissue+label%3Aaugmented-reality)

## Expected Release (Leaves Experimental)
I'm currently targeting Q2 2023.

## User Guide
The guide will be available in the app by the end of December. You can also view the guide [here](https://github.com/kylecorry31/Trail-Sense/blob/main/app/src/main/res/raw/guide_tool_augmented_reality.md)

**Reply by kylecorry31** [reply]
This feature is in Beta, and will be fully released next month.

**Comment by kylecorry31** [comment]
New experimental feature: **Tide model (auto)**

This feature enables the tide tool to lookup nearby tides completely offline without any user input (uses current location). It will be available in nightly starting tonight and will be in the 6.6.0 release behind an experimental setting (Settings > Experimental > Tide mode (auto)).

Related issues:
- https://github.com/kylecorry31/Trail-Sense/issues/2518
- https://github.com/kylecorry31/Trail-Sense/issues/1102
- https://github.com/kylecorry31/Trail-Sense/issues/2706
- https://github.com/kylecorry31/Trail-Sense/issues/2708
- https://github.com/kylecorry31/Trail-Sense/issues/2707

I plan on releasing this from experimental in the January release if all goes well (so far it seems to be pretty accurate). I will also be creating a research post about it (#1063) for those of you with a technical interest.

Known issues:
- Tides in the Gulf of Mexico / Southern US are inaccurate

**Reply by kylecorry31** [reply]
This will actually be going out in the December release as non-experimental. Tested it across 27 locations worldwide and 90% had under 45 minutes of error (50% had under 15 minutes).

**Reply by drmrok** [reply]
Truly a robust app that's unmatched in it's category. I've installed it on - every one of my EUDs. You've included so many great, useful features that cannot be found in other navigation/outdoor recreation apps that it's difficult to narrow down to ".... Among my favorite features," but I am quite impressed with your focus on privacy. I believe privacy should go hand-in-hand with spending time in the backcountry, yet most apps in this category are incredibly invasive. The app seems to place privacy and it's core, with the various features built to maintain that core principle while adapting features so that the app not only remains as functional as an app that utilizes outside resources but actually presents greater functionality than it's non privacy-respecting "peers." It should be considered that even if one does not find privacy important, the resulting self-contained resources are quite important when in areas in which power sources and tower availability are scarce, and often satellites are difficult to obtain due to canopy. 

Great work. Thanks again for your dedication.

**Reply by kylecorry31** [reply]
@drmrok thank you, I really appreciate it! When I first created Trail Sense, I didn't have privacy in mind but rather a completely offline tool (which goes hand-in-hand). Over the years, I've definitely become more aware of how much other apps don't respect the privacy of users, and that has helped to solidify my decision of this app being completely offline.

**Reply by drmrok** [reply]
Absolutely.
